perma_web/perma/utils.py

```diff
@@ -562,89 +562,88 @@ def stream_warc_if_permissible(link, user):
 #
 # Webrecorder Helpers
 #
-
-def get_wr_session(request):
-    """
-    Initializes Webrecorder session or retrieves existing session.
-    """
-    wr_username = request.session.get('wr_username')
-    wr_session_cookie = request.session.get('wr_session_cookie')
-    is_new_session = False
+def ensure_session(request, session_key, user_key, login_info):
+    session_cookie = request.session.get(session_key, '')
 
     response, data = query_wr_api(
         method='post',
-        path='/auth/anon_user',
-        cookie=wr_session_cookie,
+        path='/auth/ensure_login',
+        cookie=session_cookie,
+        json=login_info,
         valid_if=lambda code, data: code == 200
     )
 
-    # If this is a new session, WR returns a new username and session cookie:
-    # Set/Update Perma's records.
-    new_username = data['user']['username']
-    new_cookie = response.cookies.get('__wr_sesh') # NB cookie only present in api response if session is new
-    if wr_username != new_username:
-        request.session['wr_username'] = new_username
-        wr_username = new_username
-        is_new_session = True
-    if new_cookie and wr_session_cookie != new_cookie:
-        request.session['wr_session_cookie'] = new_cookie
-        wr_session_cookie = new_cookie
-        is_new_session = True
+    new_session_cookie = response.cookies.get('__wr_sesh')
+    if new_session_cookie:
+        session_cookie = new_session_cookie
+
+    request.session[session_key] = session_cookie
+    if user_key:
+        request.session[user_key] = data.get('username')
+
+    return data.get('username'), session_cookie, data.get('coll_empty')
+
+def get_wr_session(request, is_private=False, collection_slug=None):
+    """
+    Initializes Webrecorder session or retrieves existing session.
+    """
+    login_info = {}
+
+    if not is_private:
+        login_info['username'] = settings.WR_PERMA_USER
+        login_info['password'] = settings.WR_PERMA_PASSWORD
+        login_info['public'] = True
+        session_key = 'wr_public_session_cookie'
+        user_key = ''
+    else:
+        session_key = 'wr_private_session_cookie'
+        user_key = 'wr_temp_username'
 
-    return wr_username, wr_session_cookie, is_new_session
+    if collection_slug:
+        login_info['title'] = collection_slug
+        login_info['external'] = True
 
+    return ensure_session(request, session_key, user_key, login_info)
```

Comment:
Let's inline the logic from `ensure_session` here rather than breaking it out into its own function.